### ISSUE TITLE
[Feature Request]: autocomplete attribute for Input and Textarea Components

### ISSUE THREAD

**Issue by ChewbaccaCookie** [SAP/ui5-webcomponents#6995]
### Feature Request Description

The autocomplete feature is a valuable functionality for enhancing the user experience and efficiency in form inputs. Currently, the UI5 Input and Textarea components lack support for the autocomplete property, which limits their usability in scenarios where autofill suggestions or custom autocomplete behavior is desired.

We kindly request the addition of the autocomplete property to the UI5 Input and Textarea components to align them with HTML5 standards and provide more flexibility to developers. This feature would enable the following benefits:

Improved User Experience: Autocomplete functionality helps users by suggesting previously entered values or providing relevant suggestions, reducing data entry effort and potential errors.

Consistency with Standard HTML Inputs: By supporting the autocomplete property, the UI5 Input and Textarea components will align with standard HTML inputs, making it easier for developers to switch between UI5 components and native HTML inputs seamlessly.

Customized Autocomplete Behavior: The autocomplete property allows developers to specify various autocomplete values, such as "name", "email", "address", or "off", to control the type of suggestions presented to users. This flexibility enables tailored autocomplete behavior based on specific input requirements.

Accessibility Enhancements: Autocomplete functionality can benefit users who rely on assistive technologies by providing suggestions that aid in completing form fields efficiently.

We believe that the addition of the autocomplete property to the UI5 Input and Textarea components will significantly enhance the usability and flexibility of these components, empowering developers to create more intuitive and efficient user interfaces.

### Proposed Solution

Allow to define a autocomplete attribute on Input and Textarea components.
Documentation: https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete

### Priority

High

### Privacy Policy

- [X] I’m not disclosing any internal or sensitive information.

**Comment by niyap** [comment]
Hello @SAP/ui5-webcomponents-topic-rl, 

Could you please take a look of this feature request?

Kind Regards,
Niya

**Comment by hristop** [comment]
Hi @ChewbaccaCookie ,

We have done research regarding this attribute and we came to the conclusion that there are numerous issues that can't be easily resolved, besides the accessibility complexity which it brings:

- **Inconsistent Behavior**: Different browsers may interpret the autocomplete attribute differently, leading to inconsistent behavior across platforms. For example, some browsers may ignore the autocomplete attribute altogether or have their own auto-fill mechanisms that override the specified behavior.

- **Limited Control**: While the autocomplete attribute allows you to provide hints to the browser about the expected input, you have limited control over how the browser handles auto-fill suggestions. Browsers may use their own algorithms and data sources to determine what suggestions to show, which might not always align with your intended usage.

- **User Preferences**: Many modern browsers allow users to customize their auto-fill settings. Users can choose to disable auto-fill, clear saved form data, or configure specific preferences for auto-fill behavior. These user preferences can override the autocomplete attribute settings, resulting in unexpected behavior.

- **Security and Privacy Concerns**: The autocomplete attribute can involve sensitive information such as usernames, passwords, and credit card details. If not handled carefully, enabling auto-fill can pose security and privacy risks. 

- **Accessibility**: Auto-fill suggestions provided by browsers may not be accessible to all users. Some individuals with disabilities may use assistive technologies that don't interact well with auto-fill suggestions, leading to usability issues.

These are the major concerns that we have looked into when we did the research, but there are even more. That is why we have decided that we will use "autocomplete=off" in UI5 and we will refrain from even setting this attribute for the web components ("autocomplete=off" is actually ignored by the major browsers anyways). 

That is why we will have to reject this feature request. 

Best Regards,
Hristo
